packages/ember-glimmer/tests/utils/abstract-test-case.js

```diff
@@ -148,8 +148,16 @@ export class RenderingTest extends TestCase {
     run(callback);
   }
 
-  registerHelper(name, func) {
-    this.owner.register(`helper:${name}`, helper(func));
+  registerHelper(name, funcOrClassBody) {
+    let type = typeof funcOrClassBody;
+
+    if (type === 'function') {
```

Comment:
I think we should detect instances of `Ember.Helper` and throw before this.  We do not support `Ember.Helper.extend()` style helpers, but `typeof Ember.Helper.extend() === 'function'`.